### PR TITLE
Represent cItem::m_Lore as an AStringVector

### LINKED ISSUE #3475: Represent cItem::m_Lore as an AStringVector
Currently `cItem::m_Lore` is represented as a single string, which makes it impossible for Cuberite to make available the entire character set - one character needs to be set apart as the string separator when serializing / parsing the lore.

We should instead represent the lore as a full `AStringVector`, so that the full character set can be used for the lore. This change should be done as soon as possible, while there are not too many plugins actually using the `m_Lore` member.

### PR DESCRIPTION
**Pull request by peterbell10**
Closes #3475.

I've gone for the second option suggested by @madmaxoft, keeping the old `m_Lore` name reserved as a deprecated binding.  The new version is called `m_LoreTable` though I'd happily accept a better name.

This does break setting lore from the lua constructor though because I can't see a way to override constructor bindings generated with tolua++.

### CODE REVIEW

**Comment by Seadragon91** [comment]
What's about checking the passed param to `m_Lore`? With `L.GetTypeText` you can get the type of the passed param. If it's a string print a warning and push the string into the vector.

**Comment by peterbell10** [comment]
Do you mean for the constructor?

**Comment by Seadragon91** [comment]
I mean the lua variable `m_Lore`

**Comment by peterbell10** [comment]
As @madmaxoft says in the issue:
> The problem with that is when a plugin attempts to read the value, we can't know if it expects a string or a table.

I'm not sure there's a good answer to that.

**Comment by Seadragon91** [comment]
Ah yes sorry :(

File: src/Bindings/DeprecatedBindings.cpp [hunk 1]
```diff
@@ -292,6 +292,58 @@ static int tolua_AllToLua_StringToMobType00(lua_State* tolua_S)
...
+	}
+
+	cItem * Self = nullptr;
+	L.GetStackValue(1, Self);
+
+	AString LoreString = StringJoin(Self->m_LoreTable, "`");
+
+	L.Push(LoreString);
+
+	LOGWARNING("Warning in read from variable 'cItem.m_Lore': m_Lore is deprecated. Please use m_LoreTable");
```
Comment (madmaxoft) [inline src/Bindings/DeprecatedBindings.cpp]: Too wordy. `cItem.m_Lore is deprecated, use cItem.m_LoreTable instead` is enough.

File: src/Bindings/DeprecatedBindings.cpp [hunk 2]
```diff
@@ -292,6 +292,58 @@ static int tolua_AllToLua_StringToMobType00(lua_State* tolua_S)
...
+		return 0;
+	}
+
+	cItem * Self = nullptr;
+	AString LoreString;
+	L.GetStackValues(1, Self, LoreString);
+
+	Self->m_LoreTable = StringSplit(LoreString, "`");
+
+	LOGWARNING("Warning in write to variable 'cItem.m_Lore': m_Lore is deprecated. Please use m_LoreTable");
```
Comment (madmaxoft) [inline src/Bindings/DeprecatedBindings.cpp]: Same re-wording.

**Comment by madmaxoft** [comment]
Nice going :)

**Comment by peterbell10** [comment]
Looks like this fixes #3467 as well.  Though while testing that it revealed a problem with the get bindings not being const.  It works now though.

**Comment by madmaxoft** [comment]
I'm afraid the latest PR merges have produced some merge conflicts (sorry), can you fix those please?

**Comment by peterbell10** [comment]
Sure, it's conflicting with my own PR so I would've been dealing with it either way :)
